common/flatpak-dir.c

```diff
@@ -275,8 +275,18 @@ flatpak_remote_state_ensure_summary (FlatpakRemoteState *self,
                                      GError            **error)
 {
   if (self->summary == NULL)
-    return flatpak_fail_error (error, FLATPAK_ERROR_INVALID_DATA, _("Unable to load summary from remote %s: %s"), self->remote_name,
-                               self->summary_fetch_error != NULL ? self->summary_fetch_error->message : "unknown error");
+    {
+      if (self->summary_fetch_error != NULL &&
+          g_error_matches (self->summary_fetch_error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
+        {
+          g_propagate_prefixed_error (error, g_steal_pointer (&self->summary_fetch_error),
```

Comment:
Should we steal the `summary_fetch_error` here? The other code path doesn’t steal it; it just reads the error. I think we might want to `g_error_copy (self->summary_fetch_error)` instead.